addons/account/models/partner.py

```diff
@@ -454,6 +454,24 @@ def _compute_bank_count(self):
         for partner in self:
             partner.bank_account_count = mapped_data.get(partner.id, 0)
 
+    @api.constrains('company_id')
+    def _check_company_and_invoices(self):
+        partners_with_company = self.filtered(lambda p: p.company_id)
+        if partners_with_company:
+            query = """
+                SELECT partner.id
+                  FROM res_partner partner
+                  JOIN account_move move ON move.partner_id = partner.id
+                                         AND move.company_id != partner.company_id
+                WHERE partner.id IN %s
+                    AND move.type in ('out_invoice', 'in_invoice')
```

Comment:
Why the constraint? You will have an access error no matter the type of the `account.move` if you change the company of the partner.

Reply:
I wanted to be consistent with the fix on version 12, which is based on `account_invoice`
I updated the commit, so if it's ok for you, we can merge it